test/cbmc/proofs/MQTTAgent_Terminate/MQTTAgent_Terminate_harness.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * coreMQTT-Agent v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* MQTT agent include. */
+#include "mqtt_agent.h"
+#include "mqtt_agent_cbmc_state.h"
+
+void harness()
+{
+    MQTTAgentContext_t * pMqttAgentContext;
+    CommandInfo_t * pCommandInfo;
+
+    pMqttAgentContext = allocateMqttAgentContext( NULL );
+    __CPROVER_assume( isValidMqttAgentContext( pMqttAgentContext ) );
+
+    pCommandInfo = malloc( sizeof( CommandInfo_t ) );
```

Comment:
Same comment from PR #9. I think we will need a stub for `cmdCompleteCallback`.

Reply:
MQTT_Terminate is a function which just attempts to push a command to the queue. So a non deterministic value will be fine for command callback for the proof

Reply:
After giving it second thought, it might also be a good idea to explicitly define a callback that returns a `nondet_bool()` so that it is also self-documenting.

Reply:
I am still of the opinion that it doesn't add much value to this proof as all the members in the CommanInfo_t in this proof are using only a non-deterministic value. Please let me know if you think otherwise.

Reply:
As discussed offline, added more comments about using non deterministic values for the CommandInfo_t